lib/item/item.rb

```diff
@@ -1,26 +1,37 @@
-require 'date'
-
 class Item
-  attr_accessor :author, :label, :genre
-  attr_reader :publish_date, :archived, :id
+  attr_reader :id, :genre, :author, :source, :label
+  attr_accessor :publish_date
 
-  def initialize(publish_date, id = nil, archived: false)
-    @id = id || Random.rand(1..1000)
-    @publish_date = Date.parse(publish_date)
-    @archived = archived
+  def initialize(publish_date)
+    @id = Random.rand(1..1000)
+    @publish_date = publish_date
   end
```

Comment:
I think better not to change this as it throw some error in my unit test
better to keep it as before, and it willl not also create some error in others code
```ruby
def initialize(publish_date, id = nil, archived: false)
    @id = id || Random.rand(1..1000)
    @publish_date = Date.parse(publish_date)
    @archived = archived
end
```

Reply:
@ichsansandy Ok i will change this